rust/namespace/src/utils.rs

```diff
@@ -269,165 +266,156 @@ pub fn inverse_indexed_bool_for_namespace(
     return Err(error!(ErrorCode::ArtifactNotPartOfNamespace));
 }
 
-pub fn pull_namespaces(artifact: &AccountInfo) -> Result<Option<Vec<NamespaceAndIndex>>> {
+pub fn pull_namespaces(artifact: &AccountInfo) -> Result<Vec<Pubkey>> {
     let data = artifact.data.borrow_mut();
+    let item_class: ItemClass = try_from_slice_unchecked(&data[8..])?;
 
-    if data[8] == 0 {
-        return Ok(None);
-    }
-
-    let amount = u32::from_le_bytes(*array_ref![data, 9, 4]);
-
-    let cursor: usize = 13;
+    let mut namespaces: Vec<Pubkey> = vec![];
 
-    let mut arr: Vec<NamespaceAndIndex> = vec![];
-    for _n in 0..amount {
-        let bytes = array_ref![data, cursor, NAMESPACE_AND_INDEX_SIZE];
-        let serialized: NamespaceAndIndex = try_from_slice_unchecked(bytes)?;
-        arr.push(serialized)
+    for ns in item_class.namespaces.unwrap() {
+        namespaces.push(ns.namespace);
     }
 
-    return Ok(Some(arr));
+    Ok(namespaces)
 }
```

Comment:
What can `Artifact` be? Is `ItemClass` also `Artifact`?

Reply:
Yeah, anything that is namespace-able is an artifact. So far I have only implemented ItemClass, if everyone is fine with how I did it i'll add the others